storage/replica.go

```diff
@@ -3232,6 +3238,7 @@ func (r *Replica) executeBatch(
 				} else {
 					ba.Timestamp.Forward(tErr.ActualTimestamp)
 				}
+				r.store.Clock().Update(tErr.ActualTimestamp)
```

Comment:
This shouldn't be necessary as that timestamp is necessarily one that this node has already encountered. Why'd you add it?